crates/analytics/src/refunds/accumulator.rs

```diff
@@ -34,6 +71,93 @@ pub trait RefundMetricAccumulator {
     fn collect(self) -> Self::MetricOutput;
 }
 
+pub trait RefundDistributionAccumulator {
+    type DistributionOutput;
+
+    fn add_distribution_bucket(&mut self, distribution: &RefundDistributionRow);
+
+    fn collect(self) -> Self::DistributionOutput;
+}
+
+impl RefundDistributionAccumulator for RefundReasonDistributionAccumulator {
+    type DistributionOutput = Option<Vec<ReasonsResult>>;
+
+    fn add_distribution_bucket(&mut self, distribution: &RefundDistributionRow) {
+        self.refund_reason_vec.push(RefundReasonDistributionRow {
+            count: distribution.count.unwrap_or_default(),
+            total: distribution
+                .total
+                .clone()
+                .map(|i| i.to_i64().unwrap_or_default())
+                .unwrap_or_default(),
+            refund_reason: distribution.refund_reason.clone().unwrap_or_default(),
+        })
+    }
+
+    fn collect(mut self) -> Self::DistributionOutput {
+        if self.refund_reason_vec.is_empty() {
+            None
+        } else {
+            self.refund_reason_vec.sort_by(|a, b| b.count.cmp(&a.count));
+            let mut res: Vec<ReasonsResult> = Vec::new();
+            for val in self.refund_reason_vec.into_iter() {
+                let perc = f64::from(u32::try_from(val.count).ok()?) * 100.0
+                    / f64::from(u32::try_from(val.total).ok()?);
+
+                res.push(ReasonsResult {
+                    reason: val.refund_reason,
+                    count: val.count,
+                    percentage: (perc * 100.0).round() / 100.0,
+                })
+            }
+
+            Some(res)
+        }
+    }
+}
+
+impl RefundDistributionAccumulator for RefundErrorMessageDistributionAccumulator {
+    type DistributionOutput = Option<Vec<ErrorMessagesResult>>;
+
+    fn add_distribution_bucket(&mut self, distribution: &RefundDistributionRow) {
+        self.refund_error_message_vec
+            .push(RefundErrorMessageDistributionRow {
+                count: distribution.count.unwrap_or_default(),
+                total: distribution
+                    .total
+                    .clone()
+                    .map(|i| i.to_i64().unwrap_or_default())
+                    .unwrap_or_default(),
+                refund_error_message: distribution
+                    .refund_error_message
+                    .clone()
+                    .unwrap_or("".to_string()),
```

Comment:
You can use `unwrap_or_default()` here and for usages elsewhere in this file.

Reply:
Done